LogEquinoxTraceTest.testRootLoggerNameTrace failed in I20260330-0010 build

See: https://download.eclipse.org/eclipse/downloads/drops4/I20260330-0010/reports.html#tests

https://download.eclipse.org/eclipse/downloads/drops4/I20260330-0010/testresults/html/org.eclipse.osgi.tests_ep440I-unit-win32-x86_64-java21_win32.win32.x86_64_21.html

```
	Expected to have no trace logs. expected:<0> but was:<1>

java.lang.AssertionError: Expected to have no trace logs. expected:<0> but was:<1>
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:647)
at org.eclipse.equinox.log.test.LogEquinoxTraceTest.testRootLoggerNameTrace(LogEquinoxTraceTest.java:199)
at java.base/java.lang.reflect.Method.invoke(Method.java:580) 
```

Workspace log: https://download.eclipse.org/eclipse/downloads/drops4/I20260330-0010/testresults/ep440I-unit-win32-x86_64-java21_win32.win32.x86_64_21/org.eclipse.osgi.tests.AutomatedTests.txt

The log has no start/end for tests, so no idea if it has anything useful. 

[org.eclipse.osgi.tests.AutomatedTests.zip](https://github.com/user-attachments/files/26377201/org.eclipse.osgi.tests.AutomatedTests.zip)

If the listener was notified with a log message, there should be a `System.out` for what was logged. In the workspace log I see:

```
DEBUG TEST LOGS: count=1
   DEBUG TEST LOG: FrameworkEvent STARTED
```
```
DEBUG TEST LOGS: count=1
   DEBUG TEST LOG: StartLevel: setStartLevel: 20
```
```
DEBUG TEST LOGS: count=1
   DEBUG TEST LOG: StartLevel: setStartLevel: 20
```

`setStartLevel: 20` seems to be logged by this test code, in other test cases:

```
startLevel.setStartLevel(20);
```

So I assume its this log entry due to which the test failed:

```
FrameworkEvent STARTED
```

--- a/bundles/org.eclipse.osgi.tests/META-INF/MANIFEST.MF
+++ b/bundles/org.eclipse.osgi.tests/META-INF/MANIFEST.MF
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: Core OSGi Tests
 Bundle-SymbolicName: org.eclipse.osgi.tests;singleton:=true
-Bundle-Version: 3.22.500.qualifier
+Bundle-Version: 3.22.600.qualifier
 Bundle-Vendor: Eclipse.org
 Require-Bundle: 
  org.eclipse.core.runtime;bundle-version="[3.29.0,4.0.0)",

--- a/bundles/org.eclipse.osgi.tests/pom.xml
+++ b/bundles/org.eclipse.osgi.tests/pom.xml
@@ -19,7 +19,7 @@
   </parent>
   <groupId>org.eclipse.platform</groupId>
   <artifactId>org.eclipse.osgi.tests</artifactId>
-  <version>3.22.500-SNAPSHOT</version>
+  <version>3.22.600-SNAPSHOT</version>
   <packaging>eclipse-test-plugin</packaging>
 
   <properties>

--- a/bundles/org.eclipse.osgi.tests/src/org/eclipse/equinox/log/test/LogEquinoxTraceTest.java
+++ b/bundles/org.eclipse.osgi.tests/src/org/eclipse/equinox/log/test/LogEquinoxTraceTest.java
@@ -27,6 +27,8 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 import org.eclipse.equinox.log.SynchronousLogListener;
 import org.eclipse.osgi.internal.debug.Debug;
 import org.eclipse.osgi.launch.Equinox;
@@ -39,6 +41,7 @@ import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.BundleException;
 import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkEvent;
 import org.osgi.framework.ServiceReference;
 import org.osgi.service.condition.Condition;
 import org.osgi.service.log.LogEntry;
@@ -87,16 +90,28 @@ public class LogEquinoxTraceTest extends AbstractBundleTests {
 		Map<String, Object> configuration = new HashMap<>();
 		configuration.put(Constants.FRAMEWORK_STORAGE, config.getAbsolutePath());
 		equinox = new Equinox(configuration);
-		equinox.start();
-		loggerAdmin = equinox.getBundleContext()
-				.getService(equinox.getBundleContext().getServiceReference(LoggerAdmin.class));
-		LogReaderService logReader = equinox.getBundleContext()
-				.getService(equinox.getBundleContext().getServiceReference(LogReaderService.class));
-
-		testListener = new TestListener();
-		logReader.addLogListener(testListener);
+		equinox.init();
 
 		BundleContext systemContext = equinox.getBundleContext();
+		CountDownLatch frameworkStarted = new CountDownLatch(1);
+		systemContext.addFrameworkListener((e) -> {
+			if (FrameworkEvent.STARTED == e.getType()) {
+				frameworkStarted.countDown();
+			}
+		});
+
+		equinox.start();
+		// Need to wait for the STARTED event before continuing to ensure
+		// all events are flushed before adding the test log listener
+		frameworkStarted.await(1, TimeUnit.MINUTES);
+
+		loggerAdmin = systemContext.getService(systemContext.getServiceReference(LoggerAdmin.class));
+		testListener = new TestListener();
+
+		LogReaderService logReader = systemContext
+				.getService(systemContext.getServiceReference(LogReaderService.class));
+		logReader.addLogListener(testListener);
+
 		debugOptions = systemContext.getService(systemContext.getServiceReference(DebugOptions.class));
 	}
 

